app/utils/ElasticUtils.scala

```diff
@@ -0,0 +1,107 @@
+package utils
+
+import javax.inject.Inject
+
+import com.sksamuel.elastic4s.analyzers.{ CustomAnalyzerDefinition, KeywordAnalyzer, LowercaseTokenFilter, WhitespaceTokenizer }
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.sksamuel.elastic4s.http.bulk.BulkResponse
+import com.sksamuel.elastic4s.http.index.CreateIndexResponse
+import com.sksamuel.elastic4s.http.{ HttpClient, RequestFailure, RequestSuccess }
+import com.typesafe.scalalogging.LazyLogging
+import config.ElasticSearchConfig
+import models.Business
+import models.IndexConsts._
+
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.io.Source
+import scala.util.Try
+
+case class BulkInsertException(msg: String) extends Exception
+case class CreateIndexException(msg: String) extends Exception
+
+class ElasticUtils @Inject() (elastic: HttpClient, config: ElasticSearchConfig) extends LazyLogging {
+
+  private val Delimiter = ",(?=(?:[^\\\"]*\\\"[^\\\"]*\\\")*[^\\\"]*$)" // coma, ignore quoted comas
+  private val csvFilePath = config.csvFilePath
+  private val batchSize = 6250 // 6250 gave the best results after testing between 100 and 10,000
+  private val indexType = "business"
+
+  def recreateIndex(): Unit = {
+    removeIndex() // We don't care whether this works or not
+    createNewIndex() match {
+      case Left(f: RequestFailure) =>
+        throw CreateIndexException(s"Creation of ElasticSearch index failed with status: ${f.status}")
+      case Right(s: RequestSuccess[CreateIndexResponse]) =>
+        logger.info(s"Successfully created ElasticSearch index [${config.index}]")
+    }
+  }
+
+  def removeIndex(): Unit = elastic.execute {
+    deleteIndex(config.index)
+  }.await
+
+  def createNewIndex(): Either[RequestFailure, RequestSuccess[CreateIndexResponse]] = elastic.execute {
+    createIndex(config.index).analysis(
+      CustomAnalyzerDefinition(cBiAnalyzer, WhitespaceTokenizer, LowercaseTokenFilter)
+    ).mappings(
+        mapping(cBiType).fields(
+          textField(cBiName).boost(cBiNameBoost).analyzer(cBiAnalyzer),
+          longField(cBiUprn),
+          textField(cBiPostCode).analyzer(cBiAnalyzer),
+          longField(cBiIndustryCode),
+          textField(cBiLegalStatus).analyzer(KeywordAnalyzer),
+          textField(cBiTradingStatus).analyzer(KeywordAnalyzer),
+          textField(cBiTurnover).analyzer(cBiAnalyzer),
+          textField(cBiEmploymentBand).analyzer(cBiAnalyzer),
+          textField(cBiPayeRefs).analyzer(KeywordAnalyzer),
+          longField(cBiVatRefs),
+          textField(cBiCompanyNo).analyzer(KeywordAnalyzer)
+        )
+      )
+  }.await
+
+  def insertTestData(): Unit = {
+    logger.info(s"Inserting test data [$csvFilePath] into ElasticSearch")
+    val t0 = System.currentTimeMillis()
+    val iter = Source.fromFile(csvFilePath).getLines()
+    val header = splitCsvLine(iter.next)
```

Comment:
This will fail if the csv file is empty (next should only be called on an Iterator when hasNext is true)